File: src/main/java/fr/ribesg/blob/command/bot/PingCommand.java
package fr.ribesg.blob.command.bot;

import fr.ribesg.alix.api.Channel;
import fr.ribesg.alix.api.Server;
import fr.ribesg.alix.api.Source;
import fr.ribesg.alix.api.bot.command.Command;

public class PingCommand extends Command {

   public PingCommand() {
      super("ping", new String[]{
         "Just responds to ping. Usage: ##"
      });
   }

   @Override
   public boolean exec(final Server server, final Channel channel, final Source user, final String primaryArgument, final String[] args) {
      (channel == null ? user : channel).sendMessage((channel == null ? "P" : (user.getName() + ", p")) + "ong!");
      return true;
   }
}

File: src/main/kotlin/fr/ribesg/blob/command/PingCommand.kt
package fr.ribesg.blob.command

import org.kitteh.irc.client.library.Client
import org.kitteh.irc.client.library.element.Channel
import org.kitteh.irc.client.library.element.User

/**
 * @author Ribesg
 */

public class PingCommand(prefix: Char) : Command(prefix) {

    init {
        this.usage = this.prefix + "ping"
    }

    override fun exec(client: Client, from: User, to: Channel?, primArg: String?, args: Array<String>) {
        if (to == null) {
            from.sendMessage("Pong!")
        } else {
            to.sendMessage(from.getNick() + ", pong!")
        }
    }
}
